setup.sh

```
@@ -32,14 +44,378 @@ fi
+
+    if ! command -v home-manager &>/dev/null; then
+        info "Bootstrapping home-manager..."
+        nix run "nixpkgs#home-manager" -- init --switch --flake "$HM_FLAKE#$profile"
+    else
+        info "Applying home-manager switch (profile: $profile)..."
+        home-manager switch --flake "$HM_FLAKE#$profile"
+    fi
+
+    # Persist profile choice for updown auto-updater
+    if ! grep -q "DOTFILES_PROFILE" "$BASHRC" 2>/dev/null; then
+        echo "" >> "$BASHRC"
```

Shellcheck (0.11.0)

[style] Consider using { cmd1; cmd2; } >> file instead of individual redirects.

(SC2129)